resource-definitions/template-driver/labels/config-labels.yaml

```diff
@@ -0,0 +1,18 @@
+# This "config" type Resource Definition provides the value for the sample label
+apiVersion: entity.humanitec.io/v1b1
+kind: Definition
+metadata:
+  id: app-config
+entity:
+  name: app-config
+  type: config
+  driver_type: humanitec/template
+  driver_inputs:
+    values:
+      templates:
+        # Returns a sample output named "substream_id" to be used as a label
+        outputs: |
+          substream_id: my-example-id
+  # Set matching criteria as required
+  criteria:
+    - {}
```

Comment:
The matching does work by other means as well, e.g. through `app_id`, but you're right, you will want to have a matching by `res_id` so that the `resources['config.default#app-config']` resource reference makes sense. I added the matching criteria accordingly.